### PR TITLE
Cooldown

### LINKED ISSUE #38: Handshaking can lead to sudden end of game

### PR DESCRIPTION
**Pull request by vrde**
Fix #38
Requires #75 to be merged

### CODE REVIEW

File: src/planeta/cooldown.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { getStoredValue, storeValues } from "../services/localStorage";
+import getConfig from "../config";
+
+// COOLDOWN is in seconds, here we convert it to millisecods
+// so we don't need to do extra conversions.
+const COOLDOWN = getConfig().PLANET_A.COOLDOWN * 1000;
+
+const getCooldownFor = passport => JSON.parse(getStoredValue("cooldown", passport) || "{}");
+
+export function addHandshake(myPassport, theirPassport) {
```
Comment (MaxStalker) [inline src/planeta/cooldown.js]: The method is called addHandshake, but we don't really adding anything. Also `theirPassport` argument is not used here. Was it "by design"?

File: src/planeta/FinalizeHandshake.js [hunk 2]
```diff
@@ -34,6 +34,8 @@ export default class FinalizeHandshake extends React.Component {
         strategy
       );
     } catch (err) {
+      console.log(err);
```
Comment (MaxStalker) [inline src/planeta/FinalizeHandshake.js]: We can ommit console.log here

**Review by MaxStalker — CHANGES_REQUESTED** [review]

File: src/planeta/cooldown.js (hunk 1 above)
Comment (vrde) [reply src/planeta/cooldown.js]: `theirPassport` is used :)

File: src/planeta/FinalizeHandshake.js (hunk 2 above)
Comment (vrde) [reply src/planeta/FinalizeHandshake.js]: done!

File: src/planeta/utils.js [hunk 3]
```diff
@@ -133,7 +133,11 @@ export async function finalizeHandshake(
   const theirPassport = unpackReceipt(receipt).value;
   const wait = timeLeft(passport.output.value, theirPassport);
   if (wait > 0) {
-    const e = new Error(`Wait ${Math.round(wait/1000)} seconds before handshaking with the same player.`);
+    const e = new Error(
```
Comment (MaxStalker) [inline src/planeta/utils.js]: How about
```
const period = Math.round(wait/1000)
const errorMessage = `Wait ${period} seconds before handshaking with the same player.`
throw  new Error(errorMessage)
```

File: src/planeta/cooldown.js [hunk 4]
```diff
@@ -7,14 +7,20 @@ const COOLDOWN = getConfig().PLANET_A.COOLDOWN * 1000;
 
 const getCooldownFor = passport => JSON.parse(getStoredValue("cooldown", passport) || "{}");
 
-export function addHandshake(myPassport, theirPassport) {
+export function purge(myPassport) {
```
Comment (MaxStalker) [inline src/planeta/cooldown.js:10]: Me, personally, would move purging part into a different method and call this one `getCooldown` and then call purge inside of it :)

Comment (vrde) [reply src/planeta/cooldown.js:10]: I'd keep it like this for now.

**Review by MaxStalker — APPROVED** [review]

**Pull request merged**